api/database/auth.ts

```diff
@@ -0,0 +1,20 @@
+const { initializeApp, applicationDefault, cert } = require('firebase-admin/app');
+const { getFirestore, Timestamp, FieldValue } = require('firebase-admin/firestore');
+
+var serviceAccount: object;
+
+try {
+  serviceAccount = require('../../api/database/serviceAccountKey.json');
+  console.log('Starting application using Service Account.');
+} catch (error) {
+  console.log('Starting application using GCP.');
+  serviceAccount = null;
+}
+
+initializeApp({
+  credential: serviceAccount ? cert(serviceAccount) : applicationDefault()
+});
+
+const db = getFirestore();
```

Comment:
By that do you mean to default export a function that returns `getFirestore()` and then have `const db = getFirestoreWrapper();` in every file requiring the database?

Reply:
I recommend the following:
- Wrapping initialization into a single `initializeFirebase` function
- Calling that `initializeFirebase` function in `api/server.ts` only once on server start
- Moving the actual database export to a separate module (since it's not related to initialization per se)